recipes/newrelic/memcached/rhel.yml

```diff
@@ -0,0 +1,100 @@
+name: memcached-open-source-integration
+displayName: Memcached Open Source Integration
+description: New Relic install recipe for default Memcached Open Source on-host integration (via Infra-Agent)
+repository: https://github.com/newrelic/nri-memcached
+
+installTargets:
+  - type: host
+    os: linux
+    platform: amazon
+    platformVersion: "2"
+  - type: host
+    os: linux
+    platform: "centos"
+  - type: host
+    os: linux
+    platform: "redhat"
+
+# keyword convention for dealing with search terms that could land someone on this instrumentation project
+keywords:
+  - Infrastructure
+  - Integration
+  - memcached
+
+# Examine Infrastructure events for correlated data
+processMatch:
+  - memcached
+  - memcache
+
+# Matches partial list of the Log forwarding parameters
+# https://docs.newrelic.com/docs/logs/enable-log-management-new-relic/enable-log-monitoring-new-relic/forward-your-logs-using-infrastructure-agent#parameters
+# logMatch:
+#   - name: memcached
+#     file: /var/log/memcached/*access_log
+
+validationNrql: "SELECT count(*) from MemcachedSample where hostname like '{{.HOSTNAME}}%' FACET entityGuid SINCE 10 minutes ago"
+
+inputVars:
+  - name: "NR_CLI_HOSTNAME"
+    prompt: "Memcached Hostname (default: localhost)"
+    default: "localhost"
+  - name: "NR_CLI_PORT"
+    prompt: "Memcached Port (default: 11211)"
+    default: "11211"
+  - name: "NR_CLI_USERNAME"
+    prompt: "SASL Username (if applicable)"
+  - name: "NR_CLI_PASSWORD"
+    prompt: "SASL Password (if applicable)"
+    secret: true
```

Comment:
@setamyDG Since these are optional, we should check to see if these are empty (and if so, omit them from memcached-config.yml altogether).